File: android/app/src/main/java/online/db1k/safering/android/data/remote/models/CheckResponse.kt
package online.db1k.safering.android.data.remote.models

import com.google.gson.annotations.SerializedName

/**
 * Response from GET /v1/check?hash=<hmac-sha256>
 * Mirrors the iOS CheckResponse.swift exactly.
 * The hash field is an HMAC-SHA256 hash, not plain SHA-256.
 */
data class CheckResponse(
    @SerializedName("hash") val hash: String,
    @SerializedName("risk") val risk: Double,
    @SerializedName("label") val label: String?,
    @SerializedName("confidence") val confidence: Double,
    @SerializedName("tags") val tags: List<String>,
    @SerializedName("first_reported_at") val firstReportedAt: Long?,
    @SerializedName("report_count") val reportCount: Int,
    @SerializedName("is_confirmed") val isConfirmed: Boolean,
    @SerializedName("suggested_action") val suggestedAction: String?
)

File: ios/SafeRing/Data/Remote/Models/CheckResponse.swift
import Foundation

/// Remote API models for the SafeRing scam check response.
///
/// # Security
/// The server receives only HMAC-SHA256 hashes of phone numbers.
/// HMAC uses a per-install secret key provisioned at enrollment,
/// making the hash computationally infeasible to reverse.
/// No plaintext phone numbers are ever sent to the API.
///
/// Response from GET /v1/check?hash=<hmac-sha256>
/// Contains risk assessment for a hashed phone number.
struct CheckResponse: Decodable {
    let hash: String
    let risk: Double
    let label: String?
    let confidence: Double
    let tags: [String]
    let firstReportedAt: TimeInterval?
    let reportCount: Int
    let isConfirmed: Bool
    let suggestedAction: String?

    enum CodingKeys: String, CodingKey {
        case hash, risk, label, confidence, tags
        case firstReportedAt = "first_reported_at"
        case reportCount = "report_count"
        case isConfirmed = "is_confirmed"
        case suggestedAction = "suggested_action"
    }
}
